### PR TITLE
feat(2796):  permission settings for multichain, behind feature flag

### PR DESCRIPTION
**Pull request by EtherWizard33**
## **Description**
Hidden behind a feature flag, this is a new section in settings with the purpose of displaying all permissions, from in app dapp browser, wallet connect and SDK. This is a mocked UI version so that this part is done and available.

## **Related issues**

Contributes to solve issue: https://github.com/MetaMask/MetaMask-planning/issues/2796

## **Manual testing steps**

1. Go to settings by clicking the settings button in the bottom bar, then scroll to the Permissions item in the settings list.
2. After clicking the list item, the list of mocked permissions should be displayed.
5. These permissions present two dapps, uniswap and pancake swap, each connected via 3 different sources (Wallet Browser, SDK, and Wallet Connect). The connection sources are represented by tags in each list item.

## **Screenshots/Recordings**

| Before       | After        |
|--------------|--------------|
| Screenshot 2024-04-18 at 3 56 43 PM |Screenshot 2024-04-18 at 3 56 43 PM |

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/SDK/SDKSessionsManager/SDKSessionsManager.tsx [hunk 1]
```diff
@@ -156,7 +167,8 @@ const SDKSessionsManager = (props: SDKSessionsManagerProps) => {
       style={styles.wrapper}
       testID={SDKSelectorsIDs.SESSION_MANAGER_CONTAINER}
     >
-      {connectionsList.length + dappConnectionsList.length > 0
+      {connectionsList.length + dappConnectionsList.length > 0 ||
+      (isMutichainVersion1Enabled && mockPermissionItems.length)
```
Comment (tommasini) [inline app/components/Views/SDK/SDKSessionsManager/SDKSessionsManager.tsx:171]: Doesn't seem right using this mock data here

File: app/components/Views/SDK/SDKSessionsManager/SDKSessionsManager.tsx [hunk 2]
```diff
@@ -117,17 +122,23 @@ const SDKSessionsManager = (props: SDKSessionsManagerProps) => {
               connection={androidSession}
             />
           ))}
+          {isMutichainVersion1Enabled &&
+            mockPermissionItems.map((mockPermissionItem, _index) => (
```
Comment (tommasini) [inline app/components/Views/SDK/SDKSessionsManager/SDKSessionsManager.tsx:126]: It also seems weird here, hmm I see do we want to keep the mockData and replace it later? Can we leave a comment?

File: app/components/Views/SDK/SDKSessionsManager/SDKSessionsManager.tsx (hunk 1 above)
Comment (EtherWizard33) [reply app/components/Views/SDK/SDKSessionsManager/SDKSessionsManager.tsx:171]: Yes, I understand, but it's only to when the feature flag is enabled, when the feature flag is off, the empty state screen displays as usual.

It's only temporary until to get a feel for how the UI looks, until the actual data is ready, that is why we are using mock data for now.

**Review by tommasini — DISMISSED** [review]

File: app/components/Views/Settings/PermissionsSettings/PermissionsManager.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  TextVariant,
+} from '../../../../component-library/components/Texts/Text';
+import { isMutichainVersion1Enabled } from '../../../../util/networks';
+import { getNavigationOptionsTitle } from '../../../UI/Navbar';
+import PermissionItem from './PermissionItem';
+import mockPermissionItems from './PermissionItem/PermissionItem.constants';
+
+interface SDKSessionsManagerProps {
+  // TODO: Replace "any" with type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (tommasini) [inline app/components/Views/Settings/PermissionsSettings/PermissionsManager.tsx:25]: Can we get rid of this any since we refactor the component?

Comment (EtherWizard33) [reply app/components/Views/Settings/PermissionsSettings/PermissionsManager.tsx:25]: yes!

**Review by tommasini — APPROVED** [review]

**Pull request merged**